lib/spack/spack/test/abi_splicing.py

```diff
@@ -103,19 +104,29 @@ def test_splice_build_splice_node(install_specs, mutable_config):
 
 
 def test_double_splice(install_specs, mutable_config):
-    install_specs(
+    """Tests splicing two dependencies of an installed spec, for other installed specs"""
+    splice_t, splice_h, splice_z = install_specs(
         "splice-t@1 ^splice-h@1.0.0+compat ^splice-z@1.0.0+compat",
         "splice-h@1.0.2+compat ^splice-z@1.0.1+compat",
         "splice-z@1.0.2+compat",
     )
-    freeze_builds_config = _make_specs_non_buildable(["splice-t", "splice-h", "splice-z"])
-    mutable_config.set("packages", freeze_builds_config)
+    mutable_config.set("packages", _make_specs_non_buildable(["splice-t", "splice-h", "splice-z"]))
 
     goal_spec = Spec("splice-t@1 ^splice-h@1.0.2+compat ^splice-z@1.0.2+compat")
     with pytest.raises(SolverError):
         spack.concretize.concretize_one(goal_spec)
+
     _enable_splicing()
-    assert spack.concretize.concretize_one(goal_spec).satisfies(goal_spec)
+    concrete = spack.concretize.concretize_one(goal_spec)
+
+    # splice-t has a dependency that is changing, thus its hash should be different
+    assert concrete.dag_hash() != splice_t.dag_hash()
+    assert concrete.build_spec.satisfies(splice_t)
+    assert not concrete.satisfies(splice_t)
+
+    # splice-h and splice-z are reused, so the hash should stay the same
```

Comment:
`splice_h` is not reused totally, `splice_z` should be spliced into it and thus these dag_hashes should not be equal.

Reply:
Should be addressed. Note that this test is flaky, so I marked it as xfail for the time being: https://github.com/spack/spack/actions/runs/12792306051/job/35662959299?pr=48574